src/python/pants/base/exception_sink.py

```diff
@@ -212,6 +213,23 @@ def reset_log_location(cls, new_log_location: str) -> None:
         cls._pid_specific_error_fileobj = pid_specific_error_stream
         cls._shared_error_fileobj = shared_error_stream
 
+    @classmethod
+    def open_pid_specific_error_stream(cls, path):
+        ret = safe_open(path, mode="w")
+
+        def unlink_if_empty():
+            try:
+                if os.path.getsize(path) == 0:
+                    os.unlink(path)
+            except Exception:
```

Comment:
I'd only be catching the known-OK error which is OSError ENOENT myself. I'd want to know if there was any sort of other unexpected error.